tests/interfaces/test_autograd.py

```diff
@@ -811,9 +796,165 @@ def cost(a, b, c, weights):
         assert np.allclose(np.hstack(res), expected, atol=tol, rtol=0)
 
         # Check that the gradient was computed
-        # for all parameters in circuit2 (i.e., wrt=None)
-        assert spy.call_args_list[0][1]["wrt"] == None
+        # for all parameters in circuit2
+        assert circuit2.get_trainable_args() == {0, 1}
 
         # check that the parameter-shift rule was not applied
         # to the first parameter of circuit1
-        assert spy.call_args_list[1][1]["wrt"] == {1, 2}
+        assert circuit1.get_trainable_args() == {1, 2}
+
+    def test_non_diff_not_a_variable(self, capsys):
+        """Test that an argument marked as non-differentiable
+        is not wrapped as a variable."""
+        dev = qml.device("default.qubit", wires=1)
+
+        @qml.qnode(dev, interface="autograd")
+        def circuit(x, y, z):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=0)
+            qml.RZ(z, wires=0)
+            print(type(x), type(y), type(z))
+            return qml.expval(qml.PauliZ(0))
+
+        x = 1
+        y = qml.numpy.array(2, requires_grad=False)
+        z = 3
+
+        res = circuit(x, y, z)
+
+        assert circuit.get_trainable_args() == {0, 2}
+
+        assert isinstance(circuit.arg_vars[0], qml.variable.Variable)
+        assert not isinstance(circuit.arg_vars[1], qml.variable.Variable)
+        assert isinstance(circuit.arg_vars[2], qml.variable.Variable)
+
+        assert circuit.arg_vars[0] != x
+        assert circuit.arg_vars[1] == y
+        assert circuit.arg_vars[2] != z
+
+        captured = capsys.readouterr()
+        assert (
+            "<class 'pennylane.variable.Variable'> <class 'pennylane.numpy.tensor.tensor'> <class 'pennylane.variable.Variable'>"
+            in captured.out
+        )
```

Comment:
Oh, thank you for this comment, I just had a great idea! I'll put the assert statement inside the QNode, instead of doing weird printing and capturing 😆

Reply:
Done!